client/src/index.jsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import ReadMoreModal from './components/ReadMoreModal.jsx';
+import './main.css';
+
+class App extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            productName: 'Nike Blazer', //default for now
+            styleName: 'Gray/Blue', //default for now
+            productDescription: '',
+            textBlock: '',
+            productPrice: 100, //default for now
+            imageUrl: 'https://static.nike.com/a/images/t_default/eric5lwitzffpoisq0rj/blazer-mid-77-vintage-shoe-flCCX4.jpg' //default for now
+        }
+        
+    } 
+
+    componentDidMount() {
+
+        //fetching my service
+        fetch(`http://localhost:5000/product/${this.props.productId}/description`)
+        .then(response => {
+            return response.json();
+        })
+        .then(data => {
+            this.setState({
+                productDescription: data.description,
+                textBlock: data.textBlock
+            })
+
```

Comment:
I think you could combine the 2 .then blocks? Since line 25 isn't async, you could just convert to json, then setState in 1 .then.

Reply:
I see what you mean. Fetch is super weird. response.json() returns a promise. 

for example,
console.log(response.json()) //will log a promise instead of a response object

it's interesting, but i do see your intuition